File: src/backend/domain/dto/courses/exercise.py
import uuid
from domain.schema.courses.exercise import CourseExerciseBase, CourseExerciseObjectiveBase

class ExerciseObjectiveDto(CourseExerciseObjectiveBase):
    id: uuid.UUID
    objective: str
    description: str
    is_completed: bool

class ExerciseDto(CourseExerciseBase):
    id: uuid.UUID
    title: str
    summary: str
    environment_id: uuid.UUID
    objectives: list[ExerciseObjectiveDto]
    
class InternalExerciseObjectiveDto(CourseExerciseObjectiveBase):
    id: uuid.UUID
    objective: str
    description: str
    test_plan: str
    is_completed: bool
    
class InternalExerciseDto(CourseExerciseBase):
    id: uuid.UUID
    title: str
    summary: str
    objectives: list[InternalExerciseObjectiveDto]

File: src/client/src/models/dto/courses/Exercise.ts
export interface ExerciseObjective {
    id: string;
    objective: string;
    description: string;
    is_completed: boolean;
}

export interface Exercise {
    id: string;
    title: string;
    summary: string;
    environment_id: string;
    objectives: ExerciseObjective[];
}
